lib/katello/tasks/pulp3_migration_stats.rake

```diff
@@ -3,17 +3,38 @@ namespace :katello do
   task :pulp3_migration_stats => [:environment] do
     User.current = User.anonymous_admin
 
-    migrated_rpms = ::Katello::Rpm.where('pulp_id LIKE ?', '%/pulp/api/v3/content/rpm/packages/%').
+    migrated_rpm_count = ::Katello::Rpm.where('pulp_id LIKE ?', '%/pulp/api/v3/content/rpm/packages/%').
       or(::Katello::Rpm.where.not(migrated_pulp3_href: nil)).count
-    migrated_errata = ::Katello::RepositoryErratum.where.not(erratum_pulp3_href: nil).count
-    migrated_repos = ::Katello::Repository.where.not(version_href: nil).count
-    migratable_repos = ::Katello::Repository.count - ::Katello::Repository.puppet_type.count -
+    migrated_erratum_count = ::Katello::RepositoryErratum.where.not(erratum_pulp3_href: nil).count
+    migrated_repo_count = ::Katello::Repository.where.not(version_href: nil).count
+    migratable_repo_count = ::Katello::Repository.count - ::Katello::Repository.puppet_type.count -
       ::Katello::Repository.ostree_type.count - ::Katello::Repository.deb_type.count
 
+    on_demand_rpm_count = Katello::RepositoryRpm.where(:repository_id => Katello::Repository.yum_type.on_demand).
+      select(:rpm_id).uniq.count
+    on_demand_unmigrated_rpm_count = on_demand_rpm_count - migrated_rpm_count
+    immediate_unmigrated_rpm_count = ::Katello::Rpm.count - migrated_rpm_count - on_demand_unmigrated_rpm_count
+
+    # On Demand RPMs: (6.46E-04)*(#RPMs) + -3.22
+    # Immediate RPMs: (9.39E-04)*(#RPMs) + -3
+    # Repositories: 0.0746*(#Repos) + -2.07
+    migration_minutes = (0.000646 * on_demand_unmigrated_rpm_count - 3.22 +
+                         0.000943 * immediate_unmigrated_rpm_count - 3 +
+                         0.0746 * migratable_repo_count).to_i
```

Comment:
trying to understand this, is the 0.0746 meaning roughly .07 of a minute? (i.e. 4 seconds) ?

Reply:
The formulas here are linear trendlines for the data I collected. The time units for the y-axis were in minutes. 0.0746 was just the slope that came along with the trendline equation.